compaction/compaction_manager.hh

```diff
@@ -64,6 +64,15 @@ class rewrite_sstables_compaction_task_executor;
 class cleanup_sstables_compaction_task_executor;
 class validate_sstables_compaction_task_executor;
 }
+
+template<typename T>
+struct config_updater {
+    serialized_action action;
+    utils::observer<T> observer;
+
+    config_updater(utils::updateable_value<T>& cfg, std::function<future<>()> action);
+};
```

Comment:
Why do we need serialized_action? can't we just update in-place?

Anyway I see the serialized_action already exists, so it's not this patch introducing the complexity.